### ISSUE TITLE
Custom component build failing on DownloadLink.svelte

### ISSUE THREAD

**Issue by kmatzen** [gradio-app/gradio#8952]
### Describe the bug

I'm attempting to create a custom component based on the File component and I can't seem to get one part to build properly.  In particular, the build fails when the custom component takes DownloadLink as a dependency.  The sample in the docs does not have this issue and does not take DownloadLink as a dependency.

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

```bash
gradio cc create CustomFile --template File
cd customfile
gradio cc build
```

```
╭────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ 📦 Building package in customfile                                                                                                                                          │
│ 🔢 Package will use version 0.0.1 defined in pyproject.toml file. Set --bump-version to automatically bump the version number.                                             │
│ ✅ Documentation generated in demo/space.py and README.md. Pass --no-generate-docs to disable auto documentation.                                                          │
│                                                                                                                                                                            │
│ 🟠 The docs were generated in simple mode. Updating python to a version greater than 3.9 will result in richer documentation.                                              │
│ 🎨 Building frontend                                                                                                                                                       │
│ 🟥 Build failed!                                                                                                                                                           │
│ 7:53:39 AM  hmr is enabled but compilerOptions.dev is false, forcing it to true                                                                                            │
│ 7:53:39 AM  options.hot is enabled but does not work on production build, forcing it to false                                                                              │
│ 7:53:39 AM  you are building for production but compilerOptions.dev is true, forcing it to false                                                                           │
│ x Build failed in 1.52s                                                                                                                                                    │
│ node:internal/process/promises:391                                                                                                                                         │
│     triggerUncaughtException(err, true /* fromPromise */);                                                                                                                 │
│     ^                                                                                                                                                                      │
│                                                                                                                                                                            │
│   Error while preprocessing /Users/matzen/git/gradio-custom/customfile/frontend/node_modules/@gradio/wasm/svelte/DownloadLink.svelte -  Encountered type error             │
│ file: /Users/matzen/git/gradio-custom/customfile/frontend/node_modules/@gradio/wasm/svelte/DownloadLink.svelte                                                             │
│ Error: Error while preprocessing /Users/matzen/git/gradio-custom/customfile/frontend/node_modules/@gradio/wasm/svelte/DownloadLink.svelte -  Encountered type error        │
│     at throwError (/Users/matzen/git/gradio-custom/customfile/frontend/node_modules/svelte-preprocess/dist/modules/errors.js:5:11)                                         │
│     at throwTypescriptError (/Users/matzen/git/gradio-custom/customfile/frontend/node_modules/svelte-preprocess/dist/modules/errors.js:9:28)                               │
│     at transpileTs (/Users/matzen/git/gradio-custom/customfile/frontend/node_modules/svelte-preprocess/dist/transformers/typescript.js:215:47)                             │
│     at mixedImportsTranspiler (/Users/matzen/git/gradio-custom/customfile/frontend/node_modules/svelte-preprocess/dist/transformers/typescript.js:270:60)                  │
│     at transformer (/Users/matzen/git/gradio-custom/customfile/frontend/node_modules/svelte-preprocess/dist/transformers/typescript.js:344:11)                             │
│     at transform (/Users/matzen/git/gradio-custom/customfile/frontend/node_modules/svelte-preprocess/dist/autoProcess.js:46:12)                                            │
│     at async /Users/matzen/git/gradio-custom/customfile/frontend/node_modules/svelte-preprocess/dist/autoProcess.js:117:29                                                 │
│     at async script (/Users/matzen/git/gradio-custom/customfile/frontend/node_modules/svelte-preprocess/dist/autoProcess.js:147:33)                                        │
│     at async process_single_tag (file:///Users/matzen/git/gradio-custom/customfile/frontend/node_modules/svelte/src/compiler/preprocess/index.js:285:21)                   │
│     at async Promise.all (index 0) {                                                                                                                                       │
│   name: 'RollupError',                                                                                                                                                     │
│   id: '/Users/matzen/git/gradio-custom/customfile/frontend/node_modules/@gradio/wasm/svelte/DownloadLink.svelte',                                                          │
│   frame: '',                                                                                                                                                               │
│   code: 'PLUGIN_ERROR',                                                                                                                                                    │
│   plugin: 'vite-plugin-svelte',                                                                                                                                            │
│   hook: 'transform',                                                                                                                                                       │
│   watchFiles: [                                                                                                                                                            │
│     '/Users/matzen/git/gradio-custom/customfile/frontend/Index.svelte',                                                                                                    │
│     '/Users/matzen/git/gradio-custom/customfile/frontend/package.json',                                                                                                    │
│     '/Users/matzen/git/gradio-custom/customfile/frontend/node_modules/@gradio/atoms/src/index.ts',                                                                         │
│     '/Users/matzen/git/gradio-custom/customfile/frontend/shared/File.svelte',                                                                                              │
│     '/Users/matzen/git/gradio-custom/customfile/frontend/node_modules/@gradio/statustracker/index.ts',                                                                     │
│     '/Users/matzen/git/gradio-custom/customfile/frontend/shared/FileUpload.svelte',                                                                                        │
│     '/Users/matzen/git/gradio-custom/customfile/frontend/Example.svelte',                                                                                                  │
│     '/Users/matzen/git/gradio-custom/customfile/frontend/shared/FilePreview.svelte',                                                                                       │
│     '/Users/matzen/git/gradio-custom/customfile/frontend/node_modules/@gradio/upload/src/index.ts',                                                                        │
│     '/Users/matzen/git/gradio-custom/customfile/frontend/node_modules/@gradio/icons/src/index.ts',                                                                         │
│     '/Users/matzen/git/gradio-custom/customfile/frontend/shared/utils.ts',                                                                                                 │
│     '/Users/matzen/git/gradio-custom/customfile/frontend/node_modules/@gradio/wasm/svelte/index.ts',                                                                       │
│     '/Users/matzen/git/gradio-custom/customfile/frontend/node_modules/@gradio/statustracker/static/index.svelte',                                                          │
│     '/Users/matzen/git/gradio-custom/customfile/frontend/node_modules/@gradio/statustracker/static/Toast.svelte',                                                          │
│     '/Users/matzen/git/gradio-custom/customfile/frontend/node_modules/@gradio/statustracker/static/Loader.svelte',                                                         │
│     '/Users/matzen/git/gradio-custom/customfile/frontend/node_modules/@gradio/atoms/src/BlockTitle.svelte',                                                                │
│     '/Users/matzen/git/gradio-custom/customfile/frontend/node_modules/@gradio/atoms/src/BlockLabel.svelte',                                                                │
│     '/Users/matzen/git/gradio-custom/customfile/frontend/node_modules/@gradio/atoms/src/Block.svelte',                                                                     │
│     '/Users/matzen/git/gradio-custom/customfile/frontend/node_modules/@gradio/atoms/src/IconButton.svelte',                                                                │
│     '/Users/matzen/git/gradio-custom/customfile/frontend/node_modules/@gradio/atoms/src/Empty.svelte',                                                                     │
│     '/Users/matzen/git/gradio-custom/customfile/frontend/node_modules/@gradio/atoms/src/Info.svelte',                                                                      │
│     '/Users/matzen/git/gradio-custom/customfile/frontend/node_modules/@gradio/atoms/src/UploadText.svelte',                                                                │
│     '/Users/matzen/git/gradio-custom/customfile/frontend/node_modules/@gradio/atoms/src/Toolbar.svelte',                                                                   │
│     '/Users/matzen/git/gradio-custom/customfile/frontend/node_modules/@gradio/atoms/src/ShareButton.svelte',                                                               │
│     '/Users/matzen/git/gradio-custom/customfile/frontend/node_modules/@gradio/atoms/src/SelectSource.svelte',                                                              │
│     '/Users/matzen/git/gradio-custom/customfile/frontend/node_modules/@gradio/statustracker/static/utils.ts',                                                              │
│     '/Users/matzen/git/gradio-custom/customfile/frontend/node_modules/@gradio/statustracker/static/ToastContent.svelte',                                                   │
│     '/Users/matzen/mambaforge/envs/gradio/lib/python3.9/site-packages/gradio/templates/frontend/assets/svelte/svelte-submodules.js',                                       │
│     '/Users/matzen/git/gradio-custom/customfile/frontend/node_modules/@gradio/utils/src/index.ts',                                                                         │
│     '/Users/matzen/git/gradio-custom/customfile/frontend/node_modules/@gradio/upload/src/Upload.svelte',                                                                   │
│     '/Users/matzen/git/gradio-custom/customfile/frontend/node_modules/@gradio/upload/src/ModifyUpload.svelte',                                                             │
│     '/Users/matzen/git/gradio-custom/customfile/frontend/node_modules/@gradio/wasm/svelte/context.ts',                                                                     │
│     '/Users/matzen/git/gradio-custom/customfile/frontend/node_modules/@gradio/wasm/svelte/file-url.ts',                                                                    │
│     '/Users/matzen/git/gradio-custom/customfile/frontend/node_modules/@gradio/wasm/svelte/DownloadLink.svelte'                                                             │
│   ],                                                                                                                                                                       │
│   [Symbol(augmented)]: true                                                                                                                                                │
│ }                                                                                                                                                                          │
│                                                                                                                                                                            │
│ Node.js v20.16.0                                                                                                                                                           │
│                                                                                                                                                                            │
│ vite v5.3.5 building for production...                                                                                                                                     │
│ transforming...                                                                                                                                                            │
│ error TS5069: Option 'declarationMap' cannot be specified without specifying option 'declaration' or option 'composite'.                                     │
│                                                                                                                                                                            │
│ ✓ 29 modules transformed.                                                                                                                                                  │
│                                                                                                                                                                            │
╰────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
```

I can get my CustomFile to build if I remove all instances of DownloadLink, Upload, and ModifyUpload.  I was also able to get my CustomFile to build and work properly if I copied the source files from @gradio/upload and @gradio/wasm/svelte into my custom component, then adjusted the imports.

### System Info

```shell
Gradio Environment Information:
------------------------------
Operating System: Darwin
gradio version: 4.39.0
gradio_client version: 1.1.1

------------------------------------------------
gradio dependencies in your environment:

aiofiles: 23.2.1
anyio: 4.4.0
fastapi: 0.111.1
ffmpy: 0.4.0
gradio-client==1.1.1 is not installed.
httpx: 0.27.0
huggingface-hub: 0.24.3
importlib-resources: 6.4.0
jinja2: 3.1.4
markupsafe: 2.1.5
matplotlib: 3.9.1
numpy: 2.0.1
orjson: 3.10.6
packaging: 24.1
pandas: 2.2.2
pillow: 10.2.0
pydantic: 2.8.2
pydub: 0.25.1
python-multipart: 0.0.9
pyyaml: 6.0.1
ruff: 0.5.5
semantic-version: 2.10.0
tomlkit==0.12.0 is not installed.
typer: 0.12.3
typing-extensions: 4.12.2
urllib3: 2.2.2
uvicorn: 0.30.3
authlib; extra == 'oauth' is not installed.
itsdangerous; extra == 'oauth' is not installed.


gradio_client dependencies in your environment:

fsspec: 2024.6.1
httpx: 0.27.0
huggingface-hub: 0.24.3
packaging: 24.1
typing-extensions: 4.12.2
websockets: 11.0.3
```
```


### Severity

I can work around it

**Comment by freddyaboulton** [comment]
I believe this is fixed on main - will check once we release today or tomorrow.

**Comment by freddyaboulton** [comment]
Not fixed on 4.40.0 will look into it

**Comment by freddyaboulton** [comment]
Please set `@gradio/preview` to `0.10.2` in `frontend/package.json`. I think there is a different bug where the custom components are not using the latest versions automatically.

**Comment by freddyaboulton** [comment]
Ok it is actually fixed in 4.40.0, i just tested incorrectly.
